lib/puppet/type/archive.rb

```diff
@@ -150,6 +150,16 @@ def should_to_s(value)
     desc 'extract command group (using this option will configure the archive file permisison to 0644 so the user can read the file).'
   end
 
+  newparam(:proxy_type) do
+    desc 'proxy type (none|ftp|http|https)'
+    newvalues(:none, :ftp, :http, :https)
+    defaultto(:none)
+  end
+
+  newparam(:proxy_server) do
+    desc 'proxy address to use when accessing source'
+  end
+
```

Comment:
Still not sure I'm convinced.  But yeah, you can always override proxy_type. You may have guessed, but I was trying to debug https://github.com/voxpupuli/puppet-archive/issues/277

with wget on the command line for example
`http_proxy=http://proxy.example.com:3128 wget https://www.example.com/somefile`
the proxy in the environment variable `http_proxy` would not be used.

You'd have to do
`https_proxy=http://proxy.example.com:3128 wget https://www.example.com/somefile`
instead and the proxy/squid would tunnel the https request.  (Having it intercept https can be done but is *far* less common I think)